src/main/java/org/mockbukkit/mockbukkit/entity/EntityTypesMock.java

```diff
@@ -434,6 +434,7 @@ public <E extends Entity, M extends EntityMock> Builder register(@NotNull Class<
 			Preconditions.checkArgument(bukkitClazz.isAssignableFrom(mockClazz), "The class %s is not a subclass of %s", mockClazz, bukkitClazz);
 			Preconditions.checkArgument(mockFactory != null, "Cannot register a null mock factory");
 			Preconditions.checkArgument(!mapping.containsKey(bukkitClazz), "Cannot register type %s because it's already registered.", bukkitClazz);
+			Preconditions.checkArgument(mockClazz != PlayerMock.class, "Not allowed to register %s.", mockClazz);
```

Comment:
Should we replace this with `mockClazz.isAssignableFrom(Player.class)`?

Reply:
That is indeed nicer